api/contrib/envoy/extensions/filters/http/golang/v3alpha/golang.proto

```diff
@@ -142,44 +29,47 @@ message Config {
     OVERRIDE = 3;
   }
 
-  // library_id is a unique ID for a dynamic library file, must be unique globally.
+  // Globally unique ID for a dynamic library file.
   string library_id = 1 [(validate.rules).string = {min_len: 1}];
 
-  // Dynamic library implementing the interface of
-  // ``StreamFilter <contrib/golang/filters/http/source/go/pkg/api.StreamFilter>``.
+  // Path to a dynamic library implementing the
+  // :repo:`StreamFilter API <contrib/golang/filters/http/source/go/pkg/api.StreamFilter>`
+  // interface.
   // [#comment:TODO(wangfakang): Support for downloading libraries from remote repositories.]
   string library_path = 2 [(validate.rules).string = {min_len: 1}];
 
-  // plugin_name is the name of the go plugin, which needs to be consistent with the name
-  // registered in http::RegisterHttpFilterConfigFactory.
+  // Name of the Go plugin.
+  //
+  // This name **must** be consistent with the name registered in ``http::RegisterHttpFilterConfigFactory``.
+  //
   string plugin_name = 3 [(validate.rules).string = {min_bytes: 1}];
 
-  // plugin_config is the configuration of the go plugin, note that this configuration is
-  // only parsed in the go plugin.
+  // Configuration for the Go plugin.
+  //
+  // .. note::
+  //     This configuration is only parsed in the go plugin, and is therefore not validated
+  //     by Envoy.
+  //
   google.protobuf.Any plugin_config = 4;
 
-  // merge_policy is the merge policy configured by the go plugin.
-  // go plugin configuration supports three dimensions: the virtual host’s typed_per_filter_config,
-  // the route’s typed_per_filter_config or filter's config.
+  // Merge policy for plugin configuration.
+  //
+  // The Go plugin configuration supports three dimensions:
+  //
+  // * Virtual host’s :ref:`typed_per_filter_config <envoy_v3_api_field_config.route.v3.VirtualHost.typed_per_filter_config>`
+  // * Route’s :ref:`typed_per_filter_config <envoy_v3_api_field_config.route.v3.Route.typed_per_filter_config>`
+  // * The filter's :ref:`plugin_config <envoy_v3_api_field_extensions.filters.http.golang.v3alpha.Config.plugin_config>`
+  //
   // The meanings are as follows:
-  // MERGE_VIRTUALHOST_ROUTER_FILTER: pass all configuration into go plugin.
-  // MERGE_VIRTUALHOST_ROUTER: pass Virtual-Host and Router configuration into go plugin.
-  // OVERRIDE: override according to Router > Virtual_host > Filter priority and pass the
-  // configuration to the go plugin.
+  //
+  // :``MERGE_VIRTUALHOST_ROUTER_FILTER``: Pass all configuration into Go plugin.
```

Comment:
im not sure i follow what it means to pass all - does it mean no merging? some sort of namespacing so the plugin can see all?

Reply:
Yes. This is to let the go plugin get all the configurations.

Reply:
can you say more about how the plugin will see the configurations? i think it deserves more explanation

Reply:
The default implementation is to give all configurations to the go plugin, and the configuration merge strategy is controlled by the go plugin itself through ` httpFilterConfigParser.Merge`. We will implement a more flexible configuration strategy on the c++ side later(envoy Go extension filter). So we can first hide the `merge_policy` parameter in the document through`[#not-implemented-hide:]`.

Reply:
i dont think its hidden currently - are you proposing that we start hiding it until the implementation has been updated ?

Reply:
Yes.

Reply:
hmm -  i feel a bit hesitant about doing that if it is implemented and currently works - if there is no merge strategy it would also necessarily also mean hiding `ConfigsPerRoute` and `RouterPlugin` and/or changing/documenting how they work without merging

Reply:
Currently the config pass all (default MERGE_VIRTUALHOST_ROUTER_FILTER) to Go plugin and merge strategy is controlled by the Go plugin itself, If the go plugin implements [StreamFilterConfigParser.Merge](https://github.com/envoyproxy/envoy/blob/main/contrib/golang/filters/http/source/go/pkg/api/filter.go#L31-L34) API. In the next version, we will implement a more flexible config strategy on the c++ side later(envoy Go extension filter).

Reply:
ok - so iiuc, effectively _only_ `MERGE_VIRTUALHOST_ROUTER_FILTER` is currently implemented and i guess how the plugin accesses the config data is documented/commented elsewhere in the `StreamFilter` API 

so the plan then i reckon is to hide `merge_policy` and then document something along the lines of 

> All configurations are passed to Go plugin, see _here_ for more information on accessing config data from your plugin

Reply:
yep, you are right.

Reply:
cool, ill update on this basis then